### PR TITLE
New formula: Preview jupyter files (.ipynb) from Finder.

### PR DESCRIPTION
**Pull request by jendas1**
Hi, 
I've created simple quick look plugin that previews jupyter notebooks (know also as python notebooks. (https://jupyter.org).

### CODE REVIEW

File: Casks/jupyter-notebook-ql.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+cask :v1 => 'jupyter-notebook-ql' do
+  version '0.1'
+  sha256 :no_check
+
+  url "https://github.com/jendas1/jupyter-notebook-quick-look/releases/download/#{version}/jupyter-notebook-quick-look.qlgenerator.zip"
+  name 'Jupyter notebok Quick Look'
+  homepage 'https://github.com/jendas1/jupyter-notebook-quick-look'
+  license :gratis
```
Comment (vitorgalvao) [inline Casks/jupyter-notebook-ql.rb]: `:public_doman`, not `:gratis` (that is for closed source).

File: Casks/jupyter-notebook-ql.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+cask :v1 => 'jupyter-notebook-ql' do
+  version '0.1'
+  sha256 :no_check
```
Comment (vitorgalvao) [inline Casks/jupyter-notebook-ql.rb]: This needs to be set.

File: Casks/jupyter-notebook-ql.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+  url "https://github.com/jendas1/jupyter-notebook-quick-look/releases/download/#{version}/jupyter-notebook-quick-look.qlgenerator.zip"
+  name 'Jupyter notebok Quick Look'
+  homepage 'https://github.com/jendas1/jupyter-notebook-quick-look'
+  license :gratis
+  caveats "This plugin uses python 3 nbconvert module.\n" <<
```
Comment (vitorgalvao) [inline Casks/jupyter-notebook-ql.rb]: This should be the last thing, and be separated by empty lines. Also, the style is wrong. Please check other multi-line caveats and follow the same style.

**Comment by vitorgalvao** [comment]
We have `depends_on` for dependencies, don’t ask users to install them manually when you don’t have to.

However, I don’t really like that you’re asking so much, like specific environments, for so little (a quicklook plugin). I’m more inclined to refuse the cask, to be honest, but will have to think a bit.

**Comment by jendas1** [comment]
This quick look plugin is just wrapper for nbconvert module, so I need it.

File: Casks/jupyter-notebook-ql.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
+cask :v1 => 'jupyter-notebook-ql' do
+  version '0.1'
+  sha256 '42706eedadb56c1bbe739cef480ecc919028b2c60857dad97369577613bd67fa'
+
+  url "https://github.com/jendas1/jupyter-notebook-quick-look/releases/download/#{version}/jupyter-notebook-quick-look.qlgenerator.zip"
+  name 'Jupyter notebok Quick Look'
+  homepage 'https://github.com/jendas1/jupyter-notebook-quick-look'
+  license :public_domain
+  qlplugin 'jdupyter-notebook-quick-look.qlgenerator'
+  caveats <<-EOS.undent
```
Comment (vitorgalvao) [inline Casks/jupyter-notebook-ql.rb]: Still needs a line before.

File: Casks/jupyter-notebook-ql.rb [hunk 5]
```diff
@@ -0,0 +1,19 @@
+cask :v1 => 'jupyter-notebook-ql' do
+  version '0.1'
+  sha256 '42706eedadb56c1bbe739cef480ecc919028b2c60857dad97369577613bd67fa'
+
+  url "https://github.com/jendas1/jupyter-notebook-quick-look/releases/download/#{version}/jupyter-notebook-quick-look.qlgenerator.zip"
+  name 'Jupyter notebok Quick Look'
+  homepage 'https://github.com/jendas1/jupyter-notebook-quick-look'
+  license :public_domain
+  qlplugin 'jdupyter-notebook-quick-look.qlgenerator'
```
Comment (vitorgalvao) [inline Casks/jupyter-notebook-ql.rb]: And a line before this.

File: Casks/jupyter-notebook-ql.rb [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+  qlplugin 'jupyter-notebook-quick-look.qlgenerator'
+
+  caveats <<-EOS.undent
+  This plugin uses python 3 nbconvert module.
+  Make sure that nbconvert and python 3 is installed.
+  If not, do it.
+  For python3:
+    brew install python3
+  For nbconvert
+    pip3 instal nbconvert
```
Comment (jawshooah) [inline Casks/jupyter-notebook-ql.rb]: `instal` => `install`

**Comment by jawshooah** [comment]
For the record, I'm fine with including this as long as the `caveats` are cleaned up. Adding a `depends_on` for `python3` would be overkill, though, as there are plenty of ways to install Python (I use `pyenv`, for example).

**Comment by vitorgalvao** [comment]
Here’s what I was thinking: if you cleanup the caveat to only mention *what* is needed (and don’t specify how to install), then I’m fine with it. If you add a file to your repo with further instructions (like how to install the dependencies) then I’d also be fine with having a link to it in the caveat as well.

**Comment by vitorgalvao** [comment]
@jendas1 Change the caveat to `caveats 'You need python 3 and the nbconvert module to use this plugin.` an we can merge this in.

**Comment by vitorgalvao** [comment]
Thank you for the contribution. Fixed and merged in a different commit.

**Pull request closed without merging**